test/DebugInfo/enum.swift

```diff
@@ -59,7 +59,7 @@ public func foo(_ empty : Nothing) { }
 // CHECK-SAME:             {{.*}}identifier: "$s4enum4RoseOyxG{{z?}}D")
 enum Rose<A> {
 	case MkRose(() -> A, () -> [Rose<A>])
-  // DWARF: !DICompositeType({{.*}}name: "Rose",{{.*}}identifier: "$s4enum4RoseOyxGD")
+  // DWARF: !DICompositeType({{.*}}name: "Rose",{{.*}}flags: DIFlagFwdDecl{{.*}}identifier: "$s4enum4RoseOyxGD")
```

Comment:
This looks optional to me, that you wanted to ensure that `DIFlagFwdDecl` was present. Was the flag present before this, but not checked? Or did this change introduce this flag here?

Reply:
The DIFlagFwdDecl is actually new and introduced by this patch. The presence of DIFlagFwdDecl also implicitly indicates the absence of size, which is harder to check for.

Reply:
ah, from `createForwardDecl`